core/datatrak.c

```diff
@@ -666,9 +667,10 @@ static void wlog_compl_parser(struct memblock *wl_mem, struct dive *dt_dive, int
 	 */
 	strlong = 26;
 	if (*(runner + pos_suit)) {
-		wlog_suit = calloc(strlong, 1);
-		wlog_suit = memcpy(wlog_suit, runner + pos_suit, strlong);
-		wlog_suit = to_utf8((unsigned char *) wlog_suit);
+		char *wlog_suit_temp = calloc(strlong, 1);
```

Comment:
As above. In C++ you would simply declare `strlong` as `constexpr`, which guarantees that you can use it as an index in a static array. In C you would probably #define it and then use a static array.

I'm fine with this fix for now, but this really could use some rework.